### PR TITLE
Components + Frontend | Design update

### LINKED ISSUE #98: Implement components new designs
### Overview
The new designs are available on [Figma](https://www.figma.com/file/ZEDDdXRTJxKZH5j5OvGOn4/CURA.RUN?node-id=0%3A1).
.
The structure of the pages under `examples/frontend` stays the same so this should for the most part be about updating `theme-ui` to use the new style guide (to be extracted from the Figma).
Preferably supporting two themes (`light`/`dark`) from the beginning would be great, and to keep an eye on while updating the current component's styles; for how the code is now it shouldn't be too much overhead.

Instead of me creating an issue for each of the components that need to be updated, you can create issues when you pick up the component you are working on.

I can only think of font and theme to change as overall configurations now, but if you find something else needs to be updated feel free to create an issue for it.

---

### Issues

- [x] #97 
- [x] #106 
- [x] #118 
- [x] #96 

### Components
- [x] #103 
- [x] #108 
- [x] #109
- [x] #158 
- [x] #162 
- [x] #161
- [x] #160 
- [x] #159 
- [x] #158 
- [x] #157 
- [x] #163 

### Leave last
- [x] #164 
- [x] #104

### PR DESCRIPTION
**Pull request by achrafismyname**
I think the design is finally done 
Latest working live preview: https://frontend-git-darkmode-curaos.vercel.app/

Closes #98

### CODE REVIEW

**Comment by achrafismyname** [comment]
For reviewing, I suggest starting with `components` and see how the folder structure changed a bit, then `theme.ts` which has most styling values, then you can review each component individually.

After `components` is clear, check how `theme` and `fonts` is imported in `frontend` then how I changed the pages to rely on layouts instead of eaching one having thier own styles. Then finalyl you can review each layout individually. 

And I think that's it, it seems like a big PR but most changes are not criticial because most changes are CSS and if the overall structure is not broken then it's all good. 

I'll also do a browser review, to check if everything looks good in all modern browsers/screens.

**Comment by achrafismyname** [comment]
# Description

This PR updates the design of both `components` and `frontend` by updating styles in `theme.ts` and changing the layout inside components and page so they work together properly.
I tried to make it as themable as possible, so the styles that affect the look and feel are all inside `theme.ts`. And the styles that affect the layout (order, padding, margins...) are defined inside the component itself.

Fixes #98

# Type of change

- [x] This change requires a components bump
- [x] This change affects the structure of the project

# How Has This Been Tested?
Because this change is aimed for design more than functionality, I tested like the following:
- [x] Tested on the majority of screen sizes including different resolution using DevTools and real devices
- [x] Tested on the most important browsers for compatibility, and they are all good except IE 

# Checklist:

- [x] Components | Changed the structure of the folder to export `theme.ts` and `assets`, and modified `package.json` to include assets folder in the builld package
- [x] Frontend | removed `theme.ts` and `fonts` and imported them from components instead
- [x] Components | update `theme.ts` to have all the styles from the new design (excpect base stuff that it's still imported from `tailwand`)
- [x] Components | updated each components and tested it for responsivity (the only styles defined inside component are layout related, because they are unique for each components I didn't add them to `theme.ts`)
- [x] Frontend | Changed the structure of pages so they all use layouts from `container/layout` to make it easier to maintain
- [x] Frontend | updated each layout and page to work correctly with the updated components
- [x] Frontend | made sure the website looks good on most devices
- [x] Frontend + components | for components that display images, i put a placeholder in `components` and added the images to `frontend`
- [x] Frontend + components | added `dark mode` buttons to `Header` component but the mode is handled by the frontend.

# Notes:
- Vercel deployment won't work before bumping the components
- I changed the layout in `Storybook` from `centered` to `padded` because it affects the styles of components and doesn't represent a real use case

I think that's is it, hopefully :)

**Comment by devey** [comment]
> Failed



> This pull request is being automatically deployed with Vercel ([learn more](https://vercel.link/github-learn-more)). To see the status of your deployment, click below or on the icon next to each commit.
> 
> 🔍 Inspect: https://vercel.com/curaos/frontend/AQeBs5eRdbPoXzzrecnqgFmvmjLq ✅ Preview: _Failed_

@achrafismyname any idea why it fails ?

**Comment by devey** [comment]
@achrafismyname bumped components but get this now:
```
ModuleNotFoundError: Module not found: Error: Can't resolve '@cura/components/dist/assets/fonts/index.css' in '/Users/yassinesania/Coding/Cura/examples/frontend/pages'
```

File: examples/frontend/pages/_app.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { ThemeProvider } from 'theme-ui'
 import { theme } from '../theme'
 import { RecoilRoot } from 'recoil'
 import { NearHooksProvider } from '@cura/hooks'
+import '@cura/components/dist/assets/fonts/index.css'
```
Comment (devey) [inline examples/frontend/pages/_app.tsx:7]: Isn't this better suited in `_document`?

File: examples/frontend/containers/Layouts/Bids.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// @ts-nocheck
```
Comment (devey) [inline examples/frontend/containers/Layouts/Bids.tsx:1]: I would rename `Layout` folder to `layout`

File: packages/components/package.json [hunk 3]
```diff
@@ -9,8 +9,9 @@
     ],
     "bin": {},
     "scripts": {
-        "prepack": "yarn clean && tsc",
+        "prepack": "yarn clean && tsc && yarn copy-files",
         "build": "yarn run prepack",
+        "copy-files": "copyfiles -u 1 src/assets/** dist/",
```
Comment (devey) [inline packages/components/package.json:14]: There is no better way to do this? It seems a bit like a workaround more than a permanent solution

File: examples/frontend/containers/Layout.tsx [hunk 4]
```diff
@@ -63,13 +60,16 @@ export default function Layout({ children, project = 'share' }) {
 
     useEffect(switchProject, [router.asPath])
 
+    const lastPath = router.pathname.split('/').slice(-1)[0]
```
Comment (devey) [inline examples/frontend/containers/Layout.tsx:63]: I would add a comment here to explain what's happening

**Review by devey — CHANGES_REQUESTED** [review]
Looks good! Thanks for taking care of cleaning up `/frontend.`
In the end, the important thing here was to have these dozen components to be working well, responsive, and well designed. I think we're good on that!
The rest is additional, and looking at the code, it seems like it shouldn't be too hard to move on based on decisions.
Let's figure out why it's not working on Vercel and close it!

File: packages/components/package.json (hunk 3 above)
Comment (achrafismyname) [reply packages/components/package.json:14]: There is no functionality in `tsc` to copy other files than .ts to the build directory. And `copyfiles` was the only working solution i could find.
There is an issue about this here https://github.com/Microsoft/TypeScript/issues/30835

File: examples/frontend/pages/_app.tsx (hunk 1 above)
Comment (achrafismyname) [reply examples/frontend/pages/_app.tsx:7]: I get the following error when I do that
````
CSS cannot be imported within pages/_document.js. Please move global styles to pages/_app.js.
````

File: packages/components/package.json (hunk 3 above)
Comment (devey) [reply packages/components/package.json]: Thanks for the link, this seems to have worked and deployment is done!

> honestly, I have just spent hours and hours on a very weird problem just to realize that the reason was that copyfiles does not work the same way under linux and windows.
> Therefore, I would not recommend copyfiles especially if your build script is run as part of a docker build (under linux).
> === update =====
> Well, after further testing, it seems that adding double quotes should do the trick (hopefully):
> `"copy-files": "copyfiles -u 1 \"src/**/*.js\" lib"`

**Review by devey — APPROVED** [review]
Thanks for your great work here @achrafismyname. Help me focus on the `contracts` for a while so we can put these frontends to good use! 😊

**Pull request merged**